srtcore/core.cpp

```diff
@@ -7998,15 +8082,56 @@ void CUDT::processCtrlAck(const CPacket &ctrlpkt, const steady_clock::time_point
     }
     // This check covers fields up to ACKD_BUFFERLEFT.
 
-    // Update RTT
-    // m_iRTT = ackdata[ACKD_RTT];
-    // m_iRTTVar = ackdata[ACKD_RTTVAR];
-    // XXX These ^^^ commented-out were blocked in UDT;
-    // the current RTT calculations are exactly the same as in UDT4.
-    const int rtt = ackdata[ACKD_RTT];
+    // Extract RTT estimate and RTTVar from the ACK.
+    const int rtt    = ackdata[ACKD_RTT];
+    const int rttvar = ackdata[ACKD_RTTVAR];
+
+    // Update the values of smoothed RTT and the variation in RTT samples
+    // on subsequent RTT estimates extracted from the ACK packets
+    // (during transmission).
+    if (m_bIsSmoothedRTTReset)
+    {
+        // In the case of bidirectional transmission, apply double smoothing
+        // at the sender side. rtt extracted from the ACK packet is smoothed
+        // RTT obtained at the receiver side.
+        // TODO: The case of bidirectional transmission requires further
+        // improvements and testing. Double smoothing is applied here to be
+        // consistent with the previous behavior.
+        if (m_stats.recvTotal != 0)
+        {
+            // Ignore initial values which might arrive after smoothed RTT
+            // has been reset.
+            if (rtt != INITIAL_RTT && rttvar != INITIAL_RTTVAR)
+            {
+                m_iRTTVar = avg_iir<4>(m_iRTTVar, abs(rtt - m_iRTT));
+                m_iRTT    = avg_iir<8>(m_iRTT, rtt);
+            }
+        }
+        // In the case of unidirectional transmission, extract the values of
+        // smoothed RTT and RTT variance from the ACK packet.
+        else
+        {
+            m_iRTT    = rtt;
+            m_iRTTVar = rttvar;
+        }
+    }
+    // Reset the value of smoothed RTT to the first real RTT estimate extracted
+    // from an ACK after initialization (at the beginning of a transmission).
+    // In case of resumed connection over the same network, the very first RTT
+    // value sent within an ACK will be taken from cache and equal to previous
+    // connection's final smoothed RTT value. The reception of such a value
+    // will also trigger the smoothed RTT reset at the sender side.
+    else if (rtt != INITIAL_RTT && rttvar != INITIAL_RTTVAR)
+    {
+        m_iRTT                = rtt;
+        m_iRTTVar             = rtt / 2;
```

Comment:
Why `rtt / 2` instead of `rttvar`?

```suggestion
        m_iRTTVar             = rttvar;
```